src/magic_folder/downloader.py

```diff
@@ -460,6 +467,15 @@ def mark_overwrite(self, remote_snapshot, staged_content):
 
         :return PathState: The path state of the file that was written.
         """
+        # Could be items associated with the action but then we have to define
+        # an action type to teach Eliot how to serialize the remote snapshot
+        # and the FilePath.  Probably *should* do that but just doing this for
+        # now...  Maybe it should be /easier/ to do that?
+        Message.log(
+            message_type=u"downloader:filesystem:mark-overwrite",
+            content_relpath=magic2path(remote_snapshot.name),
+            staged_content_path=staged_content.path,
+        )
```

Comment:
I added support for `FilePath` in #474 (following the suggestion in https://github.com/itamarst/eliot/issues/374#issuecomment-467999103 to use a custom JSON encoder). That doesn't help for the mangled path, though.

For that, we should do one or both of:
- stop using mangled paths except at the point of writing to a tahoe-dir node
- provide a type for relative paths, rather than just using strings everywhere.

Reply:
A type for relative paths sounds like a really good idea.